src/ducks/resource.js

```diff
@@ -97,7 +98,25 @@ function fetchTranslations(id, type, project, language) {
       const { primary_language } = project;
       const { original, translations } = filterResources(resources, primary_language);
       if (language && language.value !== primary_language) {
-        dispatch(selectTranslation(original, translations, type, language));
+        if (translations.length) {
+          dispatch(selectTranslation(original, translations, type, language));
+          dispatch({
+            type: FETCH_TRANSLATIONS_SUCCESS,
+            payload: { original, translations, loading: false }
+          });
+        } else {
+          awaitTranslations(id, type, project, language)
+          .then(([translation]) => {
+            if (translation) {
+              translations.push(translation);
+            }
+            dispatch(selectTranslation(original, translations, type, language));
+            dispatch({
```

Comment:
Probably only need to dispatch this action if the translations array has been updated.